modernizer-annotation-processor/src/main/java/org/gaul/modernizer_annotation_processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,341 @@
+    }
+
+
+    private void makeFile(
+        File file,
+        List<String> annotatedElements
+    ) {
+        Writer writer = null;
+        try {
+            writer = new BufferedWriter(new FileWriter(file));
+            for (String element : annotatedElements) {
+                writer.write(element + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (writer != null) {
+                try {
+                    writer.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    private File getOutputDirectory() {
+        try {
+            FileObject fileObjectToGetPath =
+                processingEnv.getFiler().createResource(
+                    StandardLocation.CLASS_OUTPUT,
+                    "",
+                    "dummy-file.txt");
+            File outputDir =
+                new File(fileObjectToGetPath.getName()).getParentFile();
+            return ModernizerAnnotationUtils.getOutputDir(outputDir);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Returns a formatted version of the fully qualified class name
+     * regular expression of the element to perform the necessary
+     * checks in the plugin against the ASM parsed class name.
+     * @param classElement The element of kind ElementKind.CLASS whose
+     * formatted fully qualified class name regex is to be returned
+     * @return Formatted fully qualified class name regex of the given element
+     * after formatting it the way ASM parses a class name
+     * Example:
+     * Output: "org/gaul/mypackage/ExampleClass\$TestClass(\$.+)?"
+     */
+    private String getFullClassNameRegex(Element classElement) {
+        return getFullClassName(classElement)
+            .replace('.', '/')
+            .replace("$", "\\$") + "(\\$.+)?";
+    }
+
+    /**
+     * Returns the fully qualified class name of the element
+     * of kind ElementKind.CLASS passed as a parameter.
+     * @param classElement The element of kind ElementKind.CLASS whose
+     * fully qualified class name is to be returned
+     * @return Fully qualified class name of the given element
+     * Example:
+     * Output: "org.gaul.mypackage.ExampleClass$TestClass"
+     */
+    private String getFullClassName(Element classElement) {
+        String packageName = processingEnv
+            .getElementUtils()
+            .getPackageOf(classElement)
+            .getQualifiedName()
+            .toString();
+        String packagePrefix = !packageName.isEmpty() ? packageName + "." : "";
+        String classHeader = packagePrefix + getClassName(classElement);
+        return classHeader;
+    }
+
+    /**
+     * Returns the class name of the element of kind ElementKind.CLASS
```

Comment:
Maybe just describe it as: `Returns the package-less class name of the provided element.`